model/simulationtests/flex_measures_integrated_heat_pump_grid_ac.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require 'openstudio'
+require 'lib/baseline_model'
+
+model = BaselineModel.new
+
+# make a 1 story, 100m X 50m, 10 zone core/perimeter building
+model.add_geometry({ 'length' => 100,
+                     'width' => 50,
+                     'num_floors' => 1,
+                     'floor_to_floor_height' => 4,
+                     'plenum_height' => 1,
+                     'perimeter_zone_depth' => 3 })
+
+# add windows at a 40% window-to-wall ratio
+model.add_windows({ 'wwr' => 0.4,
+                    'offset' => 1,
+                    'application_type' => 'Above Floor' })
+
+# assign constructions from a local library to the walls/windows/etc. in the model
+model.set_constructions
+
+# set whole building space type; simplified 90.1-2004 Large Office Whole Building
+model.set_space_type
+
+# add design days to the model (Chicago)
+model.add_design_days
+
+# add thermostats
+model.add_thermostats({ 'heating_setpoint' => 24,
+                        'cooling_setpoint' => 28 })
+
+# pick out on of the zone/system pairs and add a humidifier
+# In order to produce more consistent results between different runs,
+# we sort the zones by names
+zones = model.getThermalZones.sort_by { |z| z.name.to_s }
+
+zone = zones[0]
+
+air_loop = OpenStudio::Model::AirLoopHVAC.new(model)
+supplyOutletNode = air_loop.supplyOutletNode
+
+schedule = model.alwaysOnDiscreteSchedule
+fan = OpenStudio::Model::FanOnOff.new(model, schedule)
+supp_heating_coil = OpenStudio::Model::CoilHeatingElectric.new(model, schedule)
+
+heating_coil = OpenStudio::Model::CoilHeatingDXVariableSpeed.new(model)
+heating_coil_speed_1 = OpenStudio::Model::CoilHeatingDXVariableSpeedSpeedData.new(model)
+heating_coil.addSpeed(heating_coil_speed_1)
+
+cooling_coil = OpenStudio::Model::CoilCoolingDXVariableSpeed.new(model)
+cooling_coil_speed_1 = OpenStudio::Model::CoilCoolingDXVariableSpeedSpeedData.new(model)
+cooling_coil.addSpeed(cooling_coil_speed_1)
+# TODO: cooling_coil.setGridSignalSchedule(grid_signal_schedule)
+
+grid_response_cooling_coil = OpenStudio::Model::CoilCoolingDXVariableSpeed.new(model)
+grid_response_cooling_coil_speed_1 = OpenStudio::Model::CoilCoolingDXVariableSpeedSpeedData.new(model)
+grid_response_cooling_coil.addSpeed(grid_response_cooling_coil_speed_1)
+# TODO: grid_response_cooling_coil.setGridSignalSchedule(grid_signal_schedule)
+
+coil_system = OpenStudio::Model::CoilSystemIntegratedHeatPumpAirSource.new(model, cooling_coil)
+coil_system.setHeatingCoil(heating_coil)
+coil_system.setGridResponseCoolingCoil(grid_response_cooling_coil)
```

Comment:
Same comment for all files, if you're adding new methods or new objects, please exercise every setter to demonstrate the API and so we can keep testing the validity of said setters in the future. You can call the setters with the exact same value as the defaults/ctor defaults though.